src/Aspire.Dashboard/Components/Pages/TelemetryEnabledComponentBase.cs

```diff
@@ -0,0 +1,95 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Aspire.Dashboard.Telemetry;
+using Microsoft.AspNetCore.Components;
+using Microsoft.VisualStudio.Telemetry;
+
+namespace Aspire.Dashboard.Components.Pages;
+
+/// <summary>
+/// A base class for components that would like to opt-in to automatic telemetry events, capturing
+/// 1) initial render time
+/// 2) component properties on each render
+///
+/// <remarks>If overriding <see cref="OnInitializedAsync"/>, <see cref="OnParametersSetAsync"/>/<see cref="OnParametersSet"/>, or <see cref="OnAfterRenderAsync"/>/<see cref="OnAfterRender"/>,
+/// inheritors <b>must</b> the accompanying base methods for telemetry to work properly.</remarks>
+/// </summary>
+public abstract class TelemetryEnabledComponentBase : ComponentBase
```

Comment:
I rather this was modeled with composition rather than inheritance.

You can only inherit from one thing. In this situation, telemetry support feels more like adding a capabilitiy to the component rather than it being the fundamental base type. Inheritance is just being used to avoid duplication.

I think this should be an interface, similar to `IPageWithSessionAndUrlState`.

Reply:
turned into an interface!